src/android/io/branch/BranchSDK.java

```diff
@@ -701,34 +701,39 @@ public void onInitFinished(JSONObject referringParams, BranchError error) {
 
             String out;
 
-            if (error == null) {
+            if (error == null && referringParams != null) {
 
                 // params are the deep linked params associated with the link that the user clicked -> was re-directed to this app
                 // params will be empty if no data found.
-                if (referringParams != null) {
-
-                    out = String.format("DeepLinkHandler(%s)", referringParams.toString());
-
-                    webView.sendJavascript(out);
-
+                try {
+                    if (referringParams.has("+clicked_branch_link") && referringParams.getBoolean("+clicked_branch_link")) {
+                        out = String.format("DeepLinkHandler(%s)", referringParams.toString());
+                        webView.sendJavascript(out);
+                    } else if (deepLinkUrl != null) {
+                        JSONObject message = new JSONObject();
+                        
+                        message.put("error", "Not a Branch link!");
+                        message.put("url", deepLinkUrl);
+                        deepLinkUrl = null;
+                    
+                        out = String.format("NonBranchLinkHandler(\"%s\")", message.toString());
```

Comment:
Is this not a problem for anybody else?
[https://github.com/BranchMetrics/cordova-ionic-phonegap-branch-deep-linking/issues/160#issuecomment-229323708](url)